framework/src/utils/GriddedData.C

```diff
@@ -57,6 +57,65 @@ GriddedData::GriddedData(std::string file_name)
   parse(_dim, _axes, _grid, _fcn, _step, file_name);
 }
 
+GriddedData::GriddedData(std::vector<Real> x_axis,
+                         std::vector<Real> y_axis,
+                         std::vector<Real> z_axis,
+                         std::vector<Real> t_axis,
+                         std::vector<Real> fcn)
+{
+  unsigned int num_data_pt = 1;
+  int ind = 0;
+  _axes.resize(0);
+  _fcn = fcn;
+  _dim = 0;
```

Comment:
you can use initializer list syntax and make things const maybe

Reply:
we moved `_fcn` to the constructor, the other member data need to be computed.